### DISCUSSION TITLE
Vue.js adapter

### DISCUSSION THREAD

**Discussion by ivanhofer** [ivanhofer/typesafe-i18n#121] [Ideas]
I also want to bring `typesafe-i18n` to [`Vue.js`](https://vuejs.org) users.
I never have uses Vue before so I don't really know where I should start. Also I'm not really aware all the differences of version 2 and version 3. So probably two different adapter are needed.

What needs to be done:
- [x] create a basic demo app how `typesafe-i18n` could work together with Vue
- [x] check differences between version 2 and 3
- [x] write the adapter(s)

**Comment by ivanhofer** [comment]
A few ressources were named on [this reddit post](https://www.reddit.com/r/vuejs/comments/qt01kx/need_help_to_bring_i18n_library_to_vue_developers/
): 

 - https://v3.vuejs.org/guide/plugins.html
 - https://vuejs.org/v2/api/#Vue-observable
 - https://v3.vuejs.org/api/basic-reactivity.html#reactive

**Comment by ivanhofer** [comment]
official `Vue.js@v3` support was added in version `2.57.3`.
`typesafe-i18n` can be used as a [`plugin`](https://v3.vuejs.org/guide/plugins.html).
Here is an example application: https://github.com/ivanhofer/typesafe-i18n/tree/main/examples/vue

**Comment by MissedWarrior** [comment]
Hey there! first of all, your package looks amazing, really good open-source soluion. But i found that there is no Vue 2 support. Is it possible to add another adapter for Vue 2? I saw the source files for Vue 3 adapter and it looks like it's not so hard to write an adapter(except the understanding of how does it works but that the matter of time) for my use case but maybe you want to support it officially?

**Reply by ivanhofer** [reply]
Hi @MissedWarrior thanks for your kind words.
I'm personally not deep into vue development. I haven't really worked with vue 2 yet.
I'm happy if someone could figure out a way to make it work in vue 2 and create a PR.

Like you already saw, it isn't that much magic. In the end adapters are just wrappers around the core with some functionality that handles the reactivity of a given framework.

**Reply by MissedWarrior** [reply]
Oh, I see. We are currently deciding about what i18n package to pick and if we will decide to pick typesafe-i18n I'll try my best to write it and submit a PR 👍 
Anyway, thank you for response
